stats_browser/src/stats_browser.rs

```diff
@@ -339,7 +339,7 @@ impl<'a> StatsBrowser<'a> {
         }
     }
     fn pump_network(&mut self) {
-        let mut buffer: [u8; 2048] = unsafe { mem::uninitialized() };
+        let mut buffer: [u8; 2048] = unsafe { mem::MaybeUninit::uninit().assume_init() };
```

Comment:
In every place but here, I think I'd be fine with replacing `mem::uninitialized()` with `mem::zeroed()`.

For now, I can't find a way to keep these bytes uninitialized to pass them to the socket reading; the standard library does exactly what we do here while also stating that it is undefined behavior. I think I'd prefer keeping it as it is until we find a way to pass uninitialized bytes to the OS socket.

Reply:
It seems that `[0u8; 2048]` should be ok: https://godbolt.org/z/v99MoY.

By the way, it looks like both `mem::zeroed()` and `mem::uninitialized()` result in two calls to memcpy, while `[0u8; 2048]` is just one memset. Maybe just go for the latter one everywhere? :)

Reply:
Oh, with `-C opt-level=3` `mem::zeroed()` and `[0u8; 2048]` become identical and `mem::uninitialized()` becomes a no-op: https://godbolt.org/z/Yc66d7.

Reply:
https://godbolt.org/z/bEE458

You can see here that the `memset` is still present even if we read from a socket. I don't want to pay a performance cost for not using C at that place.

Reply:
It seems that the cost isn't high though: memset() for 2048 bytes takes just 39 nanoseconds on my machine: https://gist.github.com/Fireball-Teeworlds/21f126b8dbb285baf88674e91d1fca5f.

I've measured that we call pump_network() around 2000 times within 5-second period (after just starting stats_browser). That should amount to just 0.000078s or 0.0016% of CPU within those 5 seconds.

I've also created a flamegraph of a release build and couldn't find any trace of memset at all (github doesn't allow to attach the original svg :/).

![flamegraph](https://user-images.githubusercontent.com/68162181/91905878-1c7b0180-ec9f-11ea-99d3-b99955bd2e98.png)

Reply:
I've updated commits according to this finding - in case you'd also find the trade-off okay.

Reply:
I could not verify that the `memset` is called in the optimized variant of your benchmark program, neither via gdb nor via reading the assembly.

In the debug version, it outputs values below 250ns. I guess it's fine…

I'm not convinced by the flamegraph, I doubt memset will get called (unless you verified it for that release build as well), so it could be hiding somewhere.

Reply:
True, I was testing with debug version (I think I'd have to use `cargo run --release` to test with release).

The flamegraph is for the stats_browser - so there are three options:
1. memset call is so tiny that there are no samples for it (which is likely given that the probability is just 0.0016% for ~200 samples).
2. memset was optimized out (that would be the ideal scenario).
3. memset was inlined. I believe it would've been visible as an empty space above the function if this were the case and it was taking noticeable time.